# 단일 표본 t 검정

- txt 파일의 데이터를 읽어서 float형으로 형변환
- 정규성 검정
    - kstest를 활용해 정규성 검정 진행
- 단일표본 t 검정
    - p-value < 0.05 : 귀무가설 기각, 대립가설 채택, 통계적으로 유의미하다.
    - p-value > 0.05 : 귀무가설 채택, 대립가설 기각

## 정규성 검정

In [4]:
with open("datas2/성인여성_키_데이터.txt", "r", encoding="utf8") as f:
    data = f.read().split('\n')
    # print(data)
    data = list(map(float, data))
    print(data)

[150.27, 142.94, 160.99, 157.48, 153.46, 137.5, 154.94, 159.51, 171.87, 143.69, 153.65, 160.25, 153.07, 154.42, 141.21, 154.17, 162.56, 164.44, 172.36, 141.51, 169.4, 167.97, 170.26, 157.8, 167.61]


In [11]:
from scipy import stats
import numpy as np

# 평균과 표준편차 계산
mean_value = np.mean(data)
std_dev = np.std(data, ddof=0)  # 모집단 표준편차(기본 설정)

mean_value, std_dev

(np.float64(156.9332), np.float64(9.974187774450611))

## 가설 설정
- 귀무가설: 표본 데이터의 평균은 163과 같다.
- 대립가설: 표본 데이터의 평균은 163과 다르다.

## 정규성 검정
- 방법1: KS-test

In [ ]:
#  KS-test는 주어진 표본 데이터가 정규분포를 따르는지 검정한다.

# 'norm': 비교 대상이 되는 이론적인 분포를 지정 ('norm'은 정규분포)
# args=(np.mean(data), np.std(data, ddof=0)): 표본 데이터의 평균과 표준편차를 사용하여 정규분포의 모수 설정
ks_statistic, p_value = stats.kstest(data, 'norm', args=(np.mean(data), np.std(data, ddof=0)))
ks_statistic, p_value

TypeError: ndtr() takes from 1 to 2 positional arguments but 3 were given